--- a/src/__frontend_name__/vite.config.js
+++ b/src/__frontend_name__/vite.config.js
@@ -1,5 +1,6 @@
 import { fileURLToPath, URL } from 'url';
 import { sveltekit } from '@sveltejs/kit/vite';
+import { svelteTesting } from '@testing-library/svelte/vite';
 import { defineConfig } from 'vite';
 import environment from 'vite-plugin-environment';
 import dotenv from 'dotenv';
@@ -25,10 +26,14 @@ export default defineConfig({
       },
     },
   },
+  test: {
+    environment: 'jsdom',
+  },
   plugins: [
     sveltekit(),
     environment("all", { prefix: "CANISTER_" }),
     environment("all", { prefix: "DFX_" }),
+    svelteTesting(),
   ],
   resolve: {
     alias: [
